source/_posts/2018-01-16-Testing-Vuejs-in-Rails-with-Webpacker-and-Jest.md

```diff
@@ -0,0 +1,153 @@
+
+If you did tried this, go ahead and run your tests. You'll see a `SyntaxError: Unexpected token import` in the output.
+
+The problem here is that `import` is part of ECMAScript 6, so we need the help of a transpiler such as [Babel](https://babeljs.io/).
+To get it working you need to install two packages by running `yarn add --dev babel-jest babel-preset-es2015` and add the "es2015" preset to your `.babelrc` file:
+{% codeblock lang:json %}
+{
+  "presets": ["es2015",
+    ["env", {
+			...
+{% endcodeblock %}
+If you want to take it one step further, you can add [moduleDirectories](https://facebook.github.io/jest/docs/en/configuration.html#moduledirectories-array-string) to your `.package.json` so you don't have to type out the full path to your modules:
+{% codeblock lang:json %}
+"jest": {
+	...
+  "moduleDirectories": [
+    "node_modules",
+    "app/javascript"
+	]
+}
+{% endcodeblock %}
+So what we had before as
+{% codeblock lang:javascript %}
+  import MyComponent from '../../app/javascript/my_component.vue';
+{% endcodeblock %}
+can now be written as
+{% codeblock lang:javascript %}
+  import MyComponent from 'my_component.vue';
+{% endcodeblock %}
+
+## 4. Where's Vue?
+If you followed every step, you should be getting a `SyntaxError` when trying to run your tests. This means that it successfully imported your component, but it can't yet understand Vue syntax.
+
+Fortunately, we have a package that will take care of it for us, [vue-jest](https://github.com/eddyerburgh/vue-jest).
+So go ahead and run `yarn add --dev vue-jest` along with adding "moduleFileExtensions", "transform" and "mapCoverage" as explained in the [README](https://github.com/eddyerburgh/vue-jest/blob/master/README.md).
+Your `package.json` should look something like this:
+{% codeblock lang:json %}
+"jest": {
+  ...
+	"moduleFileExtensions": [
+    "js",
+    "json",
+    "vue"
+  ],
+  "transform": {
+    "^.+\\.js$": "<rootDir>/node_modules/babel-jest",
+    ".*\\.(vue)$": "<rootDir>/node_modules/vue-jest"
+  },
+  "mapCoverage": true
+}
+{% endcodeblock %}
+
+With [moduleFileExtensions](https://facebook.github.io/jest/docs/en/configuration.html#modulefileextensions-array-string) we no longer need the `.vue` when importing Single File Components. So what we had before as
+{% codeblock lang:javascript %}
+  import MyComponent from 'my_component.vue';
+{% endcodeblock %}
+can now be written as
+{% codeblock lang:javascript %}
+  import MyComponent from 'my_component';.
+{% endcodeblock %}
+You should now be able to use `import` seamlessly.
+
+The rules in the [transform](https://facebook.github.io/jest/docs/en/configuration.html#transform-object-string-string) section indicate which package is responsible for the transformation of testing files. In our case, we want `vue-jest` to handle all our `.vue` files, so they are converted to plain javascript before being handled by Jest.
+
+And it sets mapCoverage to use source maps that the transformer emits.
```

Comment:
Maybe you could add a sentence or two about what mapCoverage. At least for me, it's not very clear what it does.